collab-document/src/importer/md_importer.rs

```diff
@@ -0,0 +1,509 @@
+  match node {
+    mdast::Node::BlockQuote(quote) => Some(&quote.children),
+    mdast::Node::ListItem(list) => Some(&list.children),
+    _ => None,
+  }
+}
+
+fn process_table(document_data: &mut DocumentData, table: &mdast::Table, parent_id: &str) {
+  for (row_index, row) in table.children.iter().enumerate() {
+    if let mdast::Node::TableRow(row_node) = row {
+      for (col_index, cell) in row_node.children.iter().enumerate() {
+        if let mdast::Node::TableCell(cell_node) = cell {
+          let cell_id = generate_id();
+          let cell_block =
+            create_table_cell_block(&cell_id, parent_id, row_index, col_index, &table.align);
+
+          let paragraph_node = mdast::Node::Paragraph(mdast::Paragraph {
+            children: cell_node.children.clone(),
+            position: None,
+          });
+
+          let paragraph_block_id = generate_id();
+          let paragraph_block = create_block(
+            &paragraph_block_id,
+            &paragraph_node,
+            Some(cell_id.clone()),
+            None,
+          );
+
+          document_data
+            .blocks
+            .insert(paragraph_block.id.clone(), paragraph_block);
+          document_data.blocks.insert(cell_id.clone(), cell_block);
+          update_children_map(document_data, Some(parent_id.to_string()), &cell_id);
+          update_children_map(
+            document_data,
+            Some(cell_id.to_string()),
+            &paragraph_block_id,
+          );
+
+          process_children(
+            document_data,
+            &cell_node.children,
+            Some(paragraph_block_id.clone()),
+            None,
+          );
+        }
+      }
+    }
+  }
+}
+
+fn create_table_cell_block(
+  id: &str,
+  parent_id: &str,
+  row: usize,
+  col: usize,
+  alignments: &[AlignKind],
+) -> Block {
+  let mut cell_data = HashMap::new();
+  cell_data.insert("rowPosition".to_string(), row.into());
+  cell_data.insert("colPosition".to_string(), col.into());
+
+  if let Some(align) = alignments.get(col) {
+    let align_str = match align {
+      AlignKind::Left => "left",
+      AlignKind::Right => "right",
+      AlignKind::Center => "center",
+      _ => "left",
+    };
+    cell_data.insert("align".to_string(), Value::String(align_str.to_string()));
+  }
+
+  Block {
+    id: id.to_string(),
+    ty: "table/cell".to_string(),
+    data: cell_data,
+    parent: parent_id.to_string(),
+    children: id.to_string(),
+    external_id: Some(id.to_string()),
+    external_type: Some("text".to_string()),
+  }
+}
+#[derive(Debug, serde::Deserialize, serde::Serialize)]
+pub struct Delta {
+  ops: Vec<Operation>,
+}
+
+#[derive(Debug, serde::Deserialize, serde::Serialize, Clone)]
+pub struct Operation {
+  insert: String,
+  #[serde(skip_serializing_if = "Vec::is_empty")]
+  attributes: Vec<(String, Value)>,
+}
+
+impl From<Value> for Operation {
+  fn from(s: Value) -> Self {
+    let s = s.as_object().unwrap();
```

Comment:
be careful with the `unwrap()`.

Reply:
Data input and output are all controlled by us.

Reply:
Can impl `TryFrom` and then use `flat_map` here. We intend to not use `unwrap` 
<img width="812" alt="image" src="https://github.com/user-attachments/assets/13efa3c0-4fe7-4b01-aaf1-3a499c36771b">